src/elfpy/utils/price.py

```diff
@@ -1,49 +1,109 @@
+
+    Arguments
+    ---------
+    target_apr : float
+        Target fixed APR in decimal units (for example, 5% APR would be 0.05)
+    share_reserves : float
+        base asset reserves in the pool
+    days_remaining : float
+        Amount of days left until bond maturity
+    time_stretch : float
+        Time stretch parameter, in years
+    init_share_price : float
+        Original share price when the pool started
+    share_price : float
+        Current share price
+
+    Returns
+    -------
+    float
+        The expected amount of bonds (token asset) in the pool, given the inputs
     """
-    Returns the total liquidity in the pool in terms of base
-
-    We are using spot_price when calculating total_liquidity to convert the two tokens into the same units.
-    Otherwise we're comparing apples(base_asset_reserves in ETH) and oranges (token_asset_reserves in ptETH)
-        ptEth = 1.0 ETH at maturity ONLY
-        ptEth = 0.95 ETH ahead of time
-    Discount factor from the time value of money
-        Present Value = Future Value / (1 + r)^n
-        Future Value = Present Value * (1 + r)^n
-    The equation converts from future value to present value at the appropriate discount rate,
-    which measures the opportunity cost of getting a dollar tomorrow instead of today.
-    discount rate = (1 + r)^n
-    spot price APR = 1 / (1 + r)^n
+    bond_reserves = (share_reserves / 2) * (
+        init_share_price * (1 + target_apr * time_remaining.normalized_time) ** (1 / time_remaining.stretched_time)
+        - share_price
+    )  # y = x/2 * (u * (1 + rt)**(1/T) - c)
+    return bond_reserves
+
+
+def calc_share_reserves(
+    target_apr: float,
+    bond_reserves: float,
+    time_remaining: StretchedTime,
+    init_share_price: float = 1,
+):
+    """Returns the assumed share (i.e. base asset) reserve amounts given the bond (i.e. token asset) reserves and APR
 
     Arguments
     ---------
-    base_asset_reserves : float
-        Base reserves in the pool
-    token_asset_reserves : float
-        Bond (pt) reserves in the pool
-    spot_price : float
-        Price of bonds (pts) in terms of base
+    target_apr : float
+        Target fixed APR in decimal units (for example, 5% APR would be 0.05)
+    bond_reserves : float
+        token asset (pt) reserves in the pool
+    days_remaining : float
+        Amount of days left until bond maturity
+    time_stretch : float
+        Time stretch parameter, in years
+    init_share_price : float
+        Original share price when the pool started
+    share_price : float
+        Current share price
+
+    Returns
+    -------
+    float
+        The expected amount of base asset in the pool, calculated from the provided parameters
+    """
+    share_reserves = bond_reserves / (
+        init_share_price * (1 - target_apr * time_remaining.normalized_time) ** (1 / time_remaining.stretched_time)
+    )  # z = y / (u * (1 - rt)**(1/T))
+    return share_reserves
```

Comment:
Same as above. It probably makes sense to stub this in the base pricing model and copy this to the YieldSpace model (YieldSpace and Hyperdrive can share this).